src/group/organizationGroup/organizationGroup.controller.ts

```diff
@@ -18,28 +20,101 @@ export class OrganizationGroup {
+    const fieldsToSend = <(keyof IOrganizationGroup)[]>_.difference(ORGANIZATION_GROUP_KEYS, ORGANIZATION_GROUP_OBJECT_FIELDS);
     return _.flatMap(<IOrganizationGroup[]>organizationGroups, k => pick(k, ...fieldsToSend));
   }
 
+  /**
+   * Checks if there is a group with this hierarchy
+   * @param name Name of OrganizationGgroup
+   * @param hierarchy Hierarchy of OrganizationGgroup
+   */
   static async getOrganizationGroupByHierarchy(name: string, hierarchy: string[]): Promise<IOrganizationGroup> {
     const cond = {
       name,
       hierarchy,
     };
     const organizationGroup = await OrganizationGroup._organizationGroupRepository.findOne(cond);
-    if (!organizationGroup) return Promise.reject(new Error(`Cannot find group with name: ${name} and hierarchy: ${hierarchy}`));
+    if (!organizationGroup) return Promise.reject(new Error(`Cannot find group with name: ${name} and hierarchy: ${hierarchy.join('/')}`));
     return organizationGroup;
   }
 
+  /**
+   * Check if groups in hierarchy exist and return object with IDs 
+   * of groups if existing, otherwise null
+   * @param hierarchy hierarchy of groups to check
+   */
+  static async getIDofOrganizationGroupsInHierarchy(hierarchy: string[]) {
+    const a: IOrganizationGroup[] = await promiseAllWithFails(hierarchy.map((p, index, hierarchy) => OrganizationGroup.getOrganizationGroupByHierarchy(p, hierarchy.slice(0, index))), null);
+    const existingGroups = {};
+    for (let index = 0; index < hierarchy.length; index++) {
+      const value = a[index].id ? a[index].id : null;
+      existingGroups[hierarchy[index]] = value;
+    }
+    return existingGroups;
+  }
+
+  /**
+  * Add organizationGroup
+  * @param organizationGroup The object with details to create organizationGroup 
+  * @param parentID ID of parent of organizationGroup to insert 
+  */
   static async createOrganizationGroup(organizationGroup: IOrganizationGroup, parentID: string = undefined): Promise<IOrganizationGroup> {
+
+    const parent = parentID ? await OrganizationGroup.getOrganizationGroupOld(parentID) : null;
+    /* In case there is a parent checking that all his ancestor 
+       lives and creates a hierarchy and an ancestor */
     if (parentID) {
-      // Create group hierarchy
-      const parentsHierarchy = await OrganizationGroup.getAncestors(parentID);
-      parentsHierarchy.unshift(parentID);
-      organizationGroup.ancestors = parentsHierarchy;
-      await OrganizationGroup.getHierarchyFromAncestors(organizationGroup.id, organizationGroup);
+
+      // If the parent is not alive checks all the other ancestors
+      if (!parent.isAlive) {
+        const parentAncestors = await OrganizationGroup.getAncestors(parentID, { isAlive: false });
+        parentAncestors.unshift(parent);
+
+        // Revives all the ancestors that are not alive and returns to each parent his son 
+        parentAncestors.forEach(async (ancestor, index, parentAncestors) => {
```

Comment:
I think the "forEach" doesn't work with async functions, since it does not "await" them.
 according to [this](https://blog.lavrton.com/javascript-loops-how-to-handle-async-await-6252dd3c795)

Reply:
I'm right asyncforeach function from [here](https://codeburst.io/javascript-async-await-with-foreach-b6ba62bbf404) and used it in this case